src/Microsoft.DotNet.Interactive.Jupyter/Shell.cs

```diff
@@ -111,9 +111,18 @@ public Task StartAsync(CancellationToken cancellationToken)
                                 request,
                                 kernelIdentity);
 
-                            await _scheduler.Schedule(context);
-
-                            await context.Done();
+                            try
+                            {
+                                JupyterRequestContext.Current = context;
```

Comment:
This isn't concurrency safe. Given that this change exposes this as global state, this could become a problem.